### PR TITLE
[docs]Update deploy-bare-metal.md

### PR DESCRIPTION
**Pull request by fengtao1998**
Revise official documents to make them easy to understand and use
Update deploy-bare-metal.md

### CODE REVIEW

**Review by murong00 — APPROVED** [review]

File: site2/docs/deploy-bare-metal.md [hunk 1]
```diff
@@ -30,10 +30,10 @@ Deploying a Pulsar cluster involves doing the following (in order):
 
 ### Requirements
 
-Currently, Pulsar is available for 64-bit **macOS*, **Linux**, and **Windows**. To use Pulsar, you need to install 64-bit JRE/JDK 8 or later versions.
+Currently, Pulsar is available for 64-bit **macOS**, **Linux**, and **Windows**. To use Pulsar, you need to install 64-bit JRE/JDK 8 or later versions.
 
-> If you already have an existing ZooKeeper cluster and want to reuse it, you do not need to prepare the machines
-> for running ZooKeeper.
+> If you already have an existing ZooKeeper cluster and want to reuse it, you do not need to prepare  additional machines 
+> running ZooKeeper.
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md:36]:
```suggestion
> If you want to reuse an existing ZooKeeper cluster, you do not need to prepare  additional machines running ZooKeeper.
```

File: site2/docs/deploy-bare-metal.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ When you deploy a Pulsar cluster, keep in mind the following basic better choice
 
 #### ZooKeeper
 
-For machines running ZooKeeper, is is recommended to use less powerful machines or VMs. Pulsar uses ZooKeeper only for periodic coordination-related and configuration-related tasks, *not* for basic operations. If you run Pulsar on [Amazon Web Services](https://aws.amazon.com/) (AWS), for example, a [t2.small](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html) instance might likely suffice.
+For machines running ZooKeeper, it is recommended to use less powerful machines or VMs. Pulsar uses ZooKeeper only for periodic coordination-related and configuration-related tasks, not for basic operations. If you run Pulsar on [Amazon Web Services](https://aws.amazon.com/) (AWS), for example, a [t2.small](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html) instance might likely suffice.
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md:68]: For machines running ZooKeeper, it is recommended to use less powerful machines or VMs. Pulsar uses ZooKeeper only for periodic coordination-related and configuration-related tasks, not for basic operations. If you run Pulsar on [Amazon Web Services (AWS)](https://aws.amazon.com/), for example, a [t2.small](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html)...

Comment (Jennifer88huang) [reply site2/docs/deploy-bare-metal.md:68]: what does "instance might likely suffice" mean?

File: site2/docs/deploy-bare-metal.md [hunk 3]
```diff
@@ -76,14 +76,14 @@ For machines running a bookie and a Pulsar broker, more powerful machines are re
 
 ## Install the Pulsar binary package
 
-> You need to install the Pulsar binary package on *each machine in the cluster*, including machines running [ZooKeeper](#deploy-a-zookeeper-cluster) and [BookKeeper](#deploy-a-bookkeeper-cluster).
+> You need to install the Pulsar binary package on each machine in the cluster, including machines running [ZooKeeper](#deploy-a-zookeeper-cluster) and [BookKeeper](#deploy-a-bookkeeper-cluster).
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md:79]:
```suggestion
> You need to install the Pulsar binary package on each machine in the cluster, including machines running ZooKeeper and BookKeeper.
```

Comment (Jennifer88huang) [reply site2/docs/deploy-bare-metal.md:79]: we don't need to add link here.
Here "ZK" and "BK" are only component, we add link to how to deploy them. It's easily found in the navigation bar.

File: site2/docs/deploy-bare-metal.md [hunk 4]
```diff
@@ -76,14 +76,14 @@ For machines running a bookie and a Pulsar broker, more powerful machines are re
...
 
 To get started deploying a Pulsar cluster on bare metal, you need to download a binary tarball release in one of the following ways:
 
 * By clicking on the link below directly, which automatically triggers a download:
   * <a href="pulsar:binary_release_url" download>Pulsar {{pulsar:version}} binary release</a>
 * From the Pulsar [downloads page](pulsar:download_page_url)
-* From the Pulsar [releases page](https://github.com/apache/pulsar/releases/latest) on [GitHub](https://github.com)
+* From the Pulsar [releases page](https://github.com/apache/pulsar/releases/latest) on GitHub
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md:86]: well done~

**Review by Jennifer88huang — COMMENTED** [review]
- From L8-20, we add some tips,  is it good practice to begin a topic with lots of tips?
- From L24-27,  if they are in order, you can use order list for them.
- "Each machine in your cluster needs to have Java 8 or Java 11 installed.": it's better to use active voice, such as "You need to install Java 8 or Java 11 on each machine in your cluster". Keep in mind that the task is conducted by "users /you".

**Comment by Jennifer88huang** [comment]
@fengtao1998 Generally good job~
You can also refine the structure, and consider the following things:
- Requirements
    - Software requirements
    - Hardware requirements
[This guides](https://docs.pingcap.com/tidb/stable/hardware-and-software-requirements) can be used as reference for restructure.

2. Before deployment, there list some content on how to install Pulsar, Connector, tiered storage. Are they duplication of what's "Run Pulsar locally"? If yes, shall we reuse the content and use link instead?

File: site2/docs/deploy-bare-metal.md (hunk 2 above)
Comment (fengtao1998) [reply site2/docs/deploy-bare-metal.md:68]: It means if you are deploying pulsar on AWS, then the example provided should be suffice.

File: site2/docs/deploy-bare-metal.md [hunk 5]
```diff
@@ -7,9 +7,9 @@ sidebar_label: Bare metal
 
 > ### Tips
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:8]: > ** Tips **
Bold font distinguishes Tips from headings when using navigation bar.

File: site2/docs/deploy-bare-metal.md [hunk 6]
```diff
@@ -7,9 +7,9 @@ sidebar_label: Bare metal
...
 >
-> 1. Single-cluster Pulsar installations should be sufficient for all but the most ambitious use cases. If you are interested in experimenting with
-> Pulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you do need to run a multi-cluster Pulsar instance,
-> see the guide [here](deploy-bare-metal-multi-cluster.md).
+> 1. Single-cluster Pulsar installation should be sufficient for most use cases. If you are interested in experimenting with
+> Pulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you need to run a multi-cluster Pulsar instance,
+> see the [guide](deploy-bare-metal-multi-cluster.md).
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:12]:
```suggestion
> 1. If you are interested in testing or using Pulsar, the easiest way is to run it on a single cluster. A single-cluster Pulsar instance supports most use cases. You can also refer to this [guide](deploy-bare-metal-multi-cluster.md) to run a multi-cluster Pulsar instance.
```

The two sentences should be in one line. It's unnecessary to put them in three lines. The changes are for your reference. You can modify the sentence by yourself.

File: site2/docs/deploy-bare-metal.md [hunk 7]
```diff
@@ -7,9 +7,9 @@ sidebar_label: Bare metal
...
 >
 > 2. If you want to use all builtin [Pulsar IO](io-overview.md) connectors in your Pulsar deployment, you need to download `apache-pulsar-io-connectors`
 > package and install `apache-pulsar-io-connectors` under `connectors` directory in the pulsar directory on every broker node or on every function-worker node if you
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:15]: Please check all incorrect \n newlines. That's not necessary. Thanks.

File: site2/docs/deploy-bare-metal.md [hunk 8]
```diff
@@ -7,9 +7,9 @@ sidebar_label: Bare metal
...
 > ### Tips
 >
-> 1. Single-cluster Pulsar installations should be sufficient for all but the most ambitious use cases. If you are interested in experimenting with
-> Pulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you do need to run a multi-cluster Pulsar instance,
-> see the guide [here](deploy-bare-metal-multi-cluster.md).
+> 1. Single-cluster Pulsar installation should be sufficient for most use cases. If you are interested in experimenting with
+> Pulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you need to run a multi-cluster Pulsar instance,
+> see the [guide](deploy-bare-metal-multi-cluster.md).
 >
 > 2. If you want to use all builtin [Pulsar IO](io-overview.md) connectors in your Pulsar deployment, you need to download `apache-pulsar-io-connectors`
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:14]:
```suggestion
> 2. If you want to use all built-in [Pulsar IO](io-overview.md) connectors, you need to download `apache-pulsar-io-connectors`
```

I think we can remove "in your Pulsar deployment"... You can decide whether we omit that.

File: site2/docs/deploy-bare-metal.md (hunk 7 above)
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:15]: I suggest we use `pulsar` directory or Pulsar directory. If it's not in code, we'd better write P in Pulsar in uppercase.

File: site2/docs/deploy-bare-metal.md [hunk 9]
```diff
@@ -19,21 +19,20 @@ sidebar_label: Bare metal
 > package and install `apache-pulsar-offloaders` under `offloaders` directory in the pulsar directory on every broker node. For more details of how to configure
 > this feature, you can refer to the [Tiered storage cookbook](cookbooks-tiered-storage.md).
 
-Deploying a Pulsar cluster involves doing the following (in order):
+Deploying a Pulsar cluster involves doing the following:
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:22]:
```suggestion
Deploying a Pulsar cluster consists of the following steps:
```
For your reference.

File: site2/docs/deploy-bare-metal.md [hunk 10]
```diff
@@ -19,21 +19,20 @@ sidebar_label: Bare metal
...
 ## Preparation
 
 ### Requirements
 
-Currently, Pulsar is available for 64-bit **macOS*, **Linux**, and **Windows**. To use Pulsar, you need to install 64-bit JRE/JDK 8 or later versions.
+Currently, Pulsar is available for 64-bit **macOS**, **Linux**, and **Windows**. To use Pulsar, you need to install 64-bit JRE/JDK 8 or later versions.
 
-> If you already have an existing ZooKeeper cluster and want to reuse it, you do not need to prepare the machines
-> for running ZooKeeper.
+> You can reuse existing Zookeeper clusters.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:35]:
```suggestion
> You can reuse existing ZooKeeper clusters.
```

Do we want to add > ** Tip** above this line?

File: site2/docs/deploy-bare-metal.md [hunk 11]
```diff
@@ -44,28 +43,25 @@ To run Pulsar on bare metal, the following configuration is recommended:
 
 > **Note**
 >
-> Broker is only supported on 64-bit JVM.
-
-> If you do not have enough machines, or to try out Pulsar in cluster mode (and expand the cluster later),
+> * Broker is only supported on 64-bit JVM.
+> * If you do not have enough machines, or to try out Pulsar in cluster mode (and expand the cluster later),
 > you can deploy a full Pulsar configuration on one node, where Zookeeper, the bookie and broker are run on the same machine.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:48]:
```suggestion
> * If you do not have enough machines, or you want to test Pulsar in cluster mode (and expand the cluster later), you can deploy Pulsar completely on one node, where ZooKeeper, bookie and broker are run on the same machine.
```

'on one node' and 'on the same machine'. Is there repetition? Can you modify it?

File: site2/docs/deploy-bare-metal.md [hunk 12]
```diff
@@ -44,28 +43,25 @@ To run Pulsar on bare metal, the following configuration is recommended:
...
-
-> If you do not have a DNS server, you can use the multi-host format in the service URL instead.
-
+> * If you do not have a DNS server, you can use the multi-host format in the service URL instead.
 Each machine in your cluster needs to have [Java 8](https://adoptopenjdk.net/?variant=openjdk8) or [Java 11](https://adoptopenjdk.net/?variant=openjdk11) installed.
 
 The following is a diagram showing the basic setup:
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:52]:
```suggestion
The diagram below illustrates the basic setup of a Pulsar cluster.
```

File: site2/docs/deploy-bare-metal.md [hunk 13]
```diff
@@ -44,28 +43,25 @@ To run Pulsar on bare metal, the following configuration is recommended:
...
 
 ![alt-text](assets/pulsar-basic-setup.png)
 
-In this diagram, connecting clients need to be able to communicate with the Pulsar cluster using a single URL. In this case, `pulsar-cluster.acme.com` abstracts over all of the message-handling brokers. Pulsar message brokers run on machines alongside BookKeeper bookies; brokers and bookies, in turn, rely on ZooKeeper.
+In this diagram, connecting clients need to communicate with the Pulsar cluster using a single URL. In this case, `pulsar-cluster.acme.com` abstracts over all of the message-handling brokers. Pulsar message brokers run on machines alongside BookKeeper bookies; brokers and bookies, in turn, rely on ZooKeeper.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:56]:
```suggestion
As shown in this diagram, connecting clients communicate with the Pulsar cluster using a single URL. More precisely, the URL `pulsar-cluster.acme.com` abstracts over all message-handling brokers. Pulsar message brokers run on machines alongside BookKeeper bookies; brokers and bookies, in turn, rely on ZooKeeper.
```

I am not sure about the meaning of "abstracts over all of the message-handling brokers"

File: site2/docs/deploy-bare-metal.md [hunk 14]
```diff
@@ -44,28 +43,25 @@ To run Pulsar on bare metal, the following configuration is recommended:
...
 
 ### Hardware considerations
 
-When you deploy a Pulsar cluster, keep in mind the following basic better choices when you do the capacity planning.
+If you deploy a Pulsar cluster, keep in mind the following basic better choices when you do the capacity planning.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:60]:
```suggestion
When deploying a Pulsar cluster, keep in mind that the capacity of your deployment might benefit from the following configuration.
```

For your reference.

File: site2/docs/deploy-bare-metal.md [hunk 15]
```diff
@@ -76,14 +72,14 @@ For machines running a bookie and a Pulsar broker, more powerful machines are re
 
 ## Install the Pulsar binary package
 
-> You need to install the Pulsar binary package on *each machine in the cluster*, including machines running [ZooKeeper](#deploy-a-zookeeper-cluster) and [BookKeeper](#deploy-a-bookkeeper-cluster).
+> You need to install the Pulsar binary package on each machine in the cluster, including machines running ZooKeeper and BookKeeper.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:75]: Shall we take it out from "Tip" and present it as "requirements"?

File: site2/docs/deploy-bare-metal.md [hunk 16]
```diff
@@ -76,14 +72,14 @@ For machines running a bookie and a Pulsar broker, more powerful machines are re
...
 
 To get started deploying a Pulsar cluster on bare metal, you need to download a binary tarball release in one of the following ways:
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:77]:
```suggestion
To deploy a Pulsar cluster on bare metal, you need to first download a binary tarball release by one of the following ways.
```

File: site2/docs/deploy-bare-metal.md [hunk 17]
```diff
@@ -111,7 +107,7 @@ Directory | Contains
 
 > Since Pulsar release `2.1.0-incubating`, Pulsar provides a separate binary distribution, containing all the `builtin` connectors.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:108]: probably we should check the written form of `builtin` and keep it consistent

File: site2/docs/deploy-bare-metal.md [hunk 18]
```diff
@@ -111,7 +107,7 @@ Directory | Contains
...
 > If you want to enable those `builtin` connectors, you can follow the instructions as below; otherwise you can
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:109]:
```suggestion
> To enable the `builtin` connectors (optional), you can follow the instructions below.
```

For your reference. You might have better ideas.

File: site2/docs/deploy-bare-metal.md [hunk 19]
```diff
@@ -111,7 +107,7 @@ Directory | Contains
...
-> skip this section for now.
+> skip this section now.
 
 To get started using builtin connectors, you need to download the connectors tarball release on every broker node in one of the following ways:
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:112]: To use ...

File: site2/docs/deploy-bare-metal.md [hunk 20]
```diff
@@ -180,7 +176,7 @@ For more details of how to configure tiered storage feature, you can refer to th
 
 > If you already have an exsiting zookeeper cluster and want to use it, you can skip this section.
 
-[ZooKeeper](https://zookeeper.apache.org) manages a variety of essential coordination- and configuration-related tasks for Pulsar. To deploy a Pulsar cluster, you need to deploy ZooKeeper first (before all other components). A 3-node ZooKeeper cluster is the recommended configuration. Pulsar does not make heavy use of ZooKeeper, so more lightweight machines or VMs should suffice for running ZooKeeper.
+[ZooKeeper](https://zookeeper.apache.org) manages a variety of essential coordination-related and configuration-related tasks for Pulsar. To deploy a Pulsar cluster, you need to deploy ZooKeeper first. A 3-node ZooKeeper cluster is the recommended configuration. Pulsar does not make heavy use of ZooKeeper, so more lightweight machines or VMs should suffice for running ZooKeeper.
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:179]:
```suggestion
[ZooKeeper](https://zookeeper.apache.org) manages a variety of essential coordination- and configuration-related tasks for Pulsar. To deploy a Pulsar cluster, you need to deploy ZooKeeper first. A 3-node ZooKeeper cluster is the recommended configuration. Pulsar does not make heavy use of ZooKeeper, so the lightweight machines or VMs should suffice for running ZooKeeper.
```

File: site2/docs/deploy-bare-metal.md [hunk 21]
```diff
@@ -180,7 +176,7 @@ For more details of how to configure tiered storage feature, you can refer to th
...
 
 To begin, add all ZooKeeper servers to the configuration specified in [`conf/zookeeper.conf`](reference-configuration.md#zookeeper) (in the Pulsar directory that you create [above](#install-the-pulsar-binary-package)). The following is an example:
```
Comment (sijia-w) [inline site2/docs/deploy-bare-metal.md:181]: Can you modify this sentence?

**Comment by fengtao1998** [comment]
@Jennifer88huang PTAL

File: site2/docs/deploy-bare-metal.md [hunk 22]
```diff
@@ -5,35 +5,30 @@ sidebar_label: Bare metal
 ---
 
 
-> ### Tips
+> **Tips**
 >
-> 1. Single-cluster Pulsar installations should be sufficient for all but the most ambitious use cases. If you are interested in experimenting with
-> Pulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you do need to run a multi-cluster Pulsar instance,
-> see the guide [here](deploy-bare-metal-multi-cluster.md).
+> 1. Single-cluster Pulsar installation should be sufficient for most use cases. If you are interested in experimenting withPulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you need to run a multi-cluster Pulsar instance, see the [guide](deploy-bare-metal-multi-cluster.md).
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md:10]:
```suggestion
> 1. You can use single-cluster Pulsar installation in most use cases, such as experimenting with Pulsar or using Pulsar in a startup or in a single team. If you need to run a multi-cluster Pulsar instance, see the [guide](deploy-bare-metal-multi-cluster.md).
```

File: site2/docs/deploy-bare-metal.md (hunk 10 above)
Comment (Jennifer88huang) [reply site2/docs/deploy-bare-metal.md:31]: ZooKeeper

File: site2/docs/deploy-bare-metal.md [hunk 23]
```diff
@@ -43,29 +38,25 @@ To run Pulsar on bare metal, the following configuration is recommended:
...
-> Broker is only supported on 64-bit JVM.
-
-> If you do not have enough machines, or to try out Pulsar in cluster mode (and expand the cluster later),
-> you can deploy a full Pulsar configuration on one node, where Zookeeper, the bookie and broker are run on the same machine.
-
-> If you do not have a DNS server, you can use the multi-host format in the service URL instead.
-
+> 
+> * Broker is only supported on 64-bit JVM.
+> * If you do not have enough machines, or you want to test Pulsar in cluster mode (and expand the cluster later), You can fully deploy pulsar on a node on which zookeeper, bookie and broker run
```
Comment (Jennifer88huang) [inline site2/docs/deploy-bare-metal.md:43]:
```suggestion
> * If you do not have enough machines, or you want to test Pulsar in cluster mode (and expand the cluster later), You can fully deploy Pulsar on a node on which ZooKeeper, bookie and broker run.
```

**Review by Jennifer88huang — APPROVED** [review]

**Pull request merged**